schema_utils.py

```diff
@@ -94,6 +94,10 @@ def normalize_against_schema(
     normalized_obj = None # type: Any
 
     if schema[SCHEMA_KEY_TYPE] == SCHEMA_TYPE_BOOL:
+        if obj == 'true':
```

Comment:
Some of the arguments received from the frontend is in the string form but they must be conceptually boolean, so normalization must take care of this situation.
Eg: apply_draft in ExplorationHandler (editor.py)
Proof:
```
------------------------------------
(u'Request url = ', 'http://localhost:8181/createhandler/data/GM6aCf9oHqhb?apply_draft=true')
(u'Handler class name = ', 'ExplorationHandler')
(u'handler_args = ', {u'apply_draft': u'true'})
(u'Arguments = ', [u'apply_draft'])
Iterating over arguments...
(u'apply_draft', u'true')
(u'URL path elements = ', {u'exploration_id': 'GM6aCf9oHqhb'})
(u'Request method = ', 'GET')
(u'HANDLER_ARGS_SCHEMAS =  ', {u'GET': {u'apply_draft': {u'schema': {u'type': u'bool'}, u'default_value': False}, u'v': {u'schema': {u'type': u'int'}, u'default_value': None}}})
(u'URL_PATH_ARGS_SCHEMAS = , ', {u'exploration_id': {u'schema': {u'type': u'basestring'}}})
(u'GET_HANDLER_ERROR_RETURN_TYPE', u'json')
------------------------------------
```

Reply:
If they are originally basestring, then how were they handled initially?

Reply:
In general that shouldn't be the case right. 
So is there a mismatch in the functionality?
How will it affect the functionality once this change is added?

Reply:
It will not create any mismatch for the example that I have provided here, but for other cases we must use 'bool' type in schema by deeply analysing its usage in the codebase.
Thank you.